### PR TITLE
Add holidays command

### LINKED ISSUE #988: Holidays Command
Show holidays for a certain country using `https://date.nager.at/Api`.

`s!holidays br`
`s!holidays us`

### PR DESCRIPTION
**Pull request by fernandopiniani**
Closes #988

### CODE REVIEW

**Comment by fernandopiniani** [comment]
How can I test if my command is working as expected?

File: src/commands/utility/holidays.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        .catch(e => {
+          throw new Error(`INVALID_REQUEST: ${e.message}`)
+        })
+
+      if (holidays) {
+        return channel.send(embed
+          .setTitle(`${t('commands:holidays.title')} ${countryName} ${t('commands:in')} ${year}`)
+          .setDescription(holidays
+            .map(({ date, localName, name }) =>
+              `${moment(date).format('Do MMMM')} - ${localName} (${name})`
```
Comment (pedrofracassi) [inline src/commands/utility/holidays.js]: I'd say there's no need to add `(${name})` here, as it is in english and id wouldn't make much sense when the user is using the bot in some other language.

I also made some formatting changes

```suggestion
              `**${moment(date).format('Do MMMM')}** ${localName}`
```

**Comment by metehus** [comment]
Does it work om your local machine?

**Comment by pedrofracassi** [comment]
I recommend setting a development instance up for testing and development. If you need help with the setup, shoot me a DM in Discord, I'm **pedrofracassi#4623** there.

I'll give this branch a test when I get home later today.

File: src/commands/utility/holidays.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          ).join('\n')
+        ))
+    }
+
+    try {
+      const url = `https://date.nager.at/api/v2/publicholidays/${year}/${countryCode}`
+      const holidays = await fetch(url)
+        .then(res => res.json())
+        .catch(e => {
+          throw new Error(`INVALID_REQUEST: ${e.message}`)
```
Comment (metehus) [inline src/commands/utility/holidays.js]: throw a CommandError instead

File: src/commands/utility/holidays.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
+const { Command, CommandError, SwitchbladeEmbed } = require('../../')
+const moment = require('moment')
+const fetch = require('node-fetch')
+
+const countries = {
```
Comment (metehus) [inline src/commands/utility/holidays.js]: add country names to locales so it can be translated (countries.json?)

Comment (pedrofracassi) [reply src/commands/utility/holidays.js]: Isn't there an NPM module with all country names + translations?

cc @davipatury

File: src/commands/utility/holidays.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      const url = `https://date.nager.at/api/v2/publicholidays/${year}/${countryCode}`
+      const holidays = await fetch(url)
+        .then(res => res.json())
+        .catch(e => {
+          throw new Error(`INVALID_REQUEST: ${e.message}`)
+        })
+
+      if (holidays) {
+        return channel.send(embed
+          .setTitle(`${t('commands:holidays.title')} ${countryName} ${t('commands:in')} ${year}`)
```
[the remark this answers is not in the document]
Comment (fernandopiniani) [reply src/commands/utility/holidays.js]: Cool!

File: src/commands/utility/holidays.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    const countryName = countries[countryCode.toUpperCase()]
+    if (!countryName) {
+      return channel.send(embed
+        .setTitle(t('commands:holidays.invalidCountry'))
+        .setDescription(Object.entries(countries)
+          .map(([countryCode, countryName]) =>
+            `${countryCode} - ${countryName}`
+          ).join('\n')
+        ))
+    }
```
[the remark this answers is not in the document]
Comment (fernandopiniani) [reply src/commands/utility/holidays.js]: I made that change, cool feature. Please check if it's done right.

File: src/commands/utility/holidays.js (hunk 3 above)
Comment (metehus) [reply src/commands/utility/holidays.js]: or just get a data from somewhere

File: src/commands/utility/holidays.js [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+  async run ({ t, author, channel }, countryCode = '', year = moment().year()) {
+    const embed = new SwitchbladeEmbed(author)
+    const countryName = countries[countryCode.toUpperCase()]
+
+    try {
+      const url = `https://date.nager.at/api/v2/publicholidays/${year}/${countryCode}`
+      const holidays = await fetch(url)
+        .then(res => res.json())
+        .catch(e => {
+          throw new CommandError(`INVALID_REQUEST: ${e.message}`)
```
Comment (metehus) [inline src/commands/utility/holidays.js]: Sorry i didn't explained. CommandError will send a error embed on the channel, so the argument is the message to display on the embed sent(you did right on line 157), so please use translations to translate a specific request or just use ``commons:error``

File: src/commands/utility/holidays.js [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+      }, {
+        type: 'number',
+        required: false,
+        min: 0
+      }]
+    })
+  }
+
+  async run ({ t, author, channel }, countryCode = '', year = moment().year()) {
+    const embed = new SwitchbladeEmbed(author)
```
Comment (metehus) [inline src/commands/utility/holidays.js]: if its not much to ask, would be nice a start typing event, most commands have it

File: src/commands/utility/holidays.js (hunk 6 above)
Comment (fernandopiniani) [reply src/commands/utility/holidays.js]: LOL! You're right! My bad. I'll fix it.

File: src/commands/utility/holidays.js (hunk 7 above)
Comment (fernandopiniani) [reply src/commands/utility/holidays.js]: What do you mean by start typing event? Is this that disclaimer message below the chat saying "Member X is typing..."? That's cool, I'll include it.

Comment (pedrofracassi) [reply src/commands/utility/holidays.js]: Yup! We usually add it to commands that make web requests, cause they might take some extra time to complete. This way the user knows the bot is working, and didn't crash or something xD

Make sure to startTyping as soon as the command is ran, and stopTyping as soon as the answer is sent.

Comment (fernandopiniani) [reply src/commands/utility/holidays.js]: Done!

**Comment by fernandopiniani** [comment]
@metehus @pedrofracassi 
After many tries, I've been able to run a bot instance on my machine. The holidays command is working!!!! I checked out leet command too and found a bug... :( I'll open and issue, sorry for that.

File: src/commands/utility/holidays.js (hunk 3 above)
Comment (fernandopiniani) [reply src/commands/utility/holidays.js]: I found this package https://github.com/michaelwittig/node-i18n-iso-countries, but how can I know to which language translate?

Comment (fernandopiniani) [reply src/commands/utility/holidays.js]: update: solved with countries.json

**Comment by fernandopiniani** [comment]
Well, I did all changes you guys asked.

**Comment by metehus** [comment]
well for me its ok, on mobile now so i will approve it later on pc

**Comment by metehus** [comment]
sorry about the long time and @pedrofracassi please re-review it

**Comment by fernandopiniani** [comment]
@pedrofracassi Can you check if all your requests are okay?

**Pull request merged**